### DISCUSSION TITLE
Resolution and Performance

### DISCUSSION THREAD

**Discussion by sganz** [luni64/EncoderTool#41] [Q&A] [answered]
Just starting out on a project to make a control panel for ham radio project. I found EncoderTool and looks like it's a perfect fit and very well done.

Not having much experience with the Arduino (Nano) looking for a feel for how it would work with an optical encoder with 360 PPR using the 74HC165 multiplexer board. The Nano would only be used to capture Encoders, and set LED's state, so not much otherwise.

The idea is to use an optical encoder with 360PPR for the main tuning/frequency control, and the possibly using 6 of the more common mechanical encoders for slower and less resolution (24PPR) for other settings like Volume, etc.

Likely want to capture the switches so I'm guessing Polling mode? Would that be fast enough for a fast spin of the frequency select knob? Sorry fast is pretty subjective :) 

Thanks for any feedback before I start ordering parts!

Sandy

**Comment by luni64** [comment, marked as answer]
I think a nano might be too slow for your 360ppr + 6 low-res encoders. I did a quick experiment with a nano and the 74165 code. I measured some **160 µs** per tick. That gives you a reading frequency of 1/160E-6 = 6.25 kHz, which might be a bit slow for your 360ppm encoder. 
Any reason why you want to use that ancient processor? If your project is cost-sensitive, you could use a [SEED XIAO](https://wiki.seeedstudio.com/Seeeduino-XIAO/) (~5 EUR), which can tick in **68 µs** (14 kHz). Or, if you can spend about 30 EUR, a Teensy 4.X, which would do the same in **3 µs**.  (Depending on what you're trying to do, using a Teensy probably won't require a second processor, which might reduce the complexity and saves the cost for the second processor)

**Reply by sganz** [reply]
I will take a look at a few of the mentioned processors. I have a few of the 16mhz Nano's and a Mega 2560 board so was going to start with that and see how far I can get. I do like the Teensy 4 board, and price is not a big deal. In thinking about how things might work more efficiently I might try with interrupts on the main optical encoder and for the rest see how poling might work as they are not often changed.  Going to order some parts for the encoders and breadboard up to see how it might work, and then change boards to see how it might do.  Thanks for the info and ideas!
